File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;

import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

@RestController
@RequestMapping("/brand")
public class BrandController {

	@Reference
	private BrandService brandService;
	
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		return brandService.findAll();		
	}

	@RequestMapping("/findPage")
	public PageResult findPage(int page ,int size){
		return brandService.findPage(page, size);
	}

	@RequestMapping("/findOne")
	public TbBrand findOne(Long id){
		return brandService.findOne(id);
	}

	@RequestMapping("/save")
	public Result save(@RequestBody TbBrand tbBrand){
		try{
			brandService.save(tbBrand);
			return new Result(true,"保存成功");
		}catch (Exception e){
			e.printStackTrace();
			return new Result(false,"保存失败成功,可能已经存在该品牌"+tbBrand.getName());
		}

	}

	@RequestMapping("/delete")
	public Result delete(Long [] ids){
		try{

			brandService.delete(ids);
			return new Result(true,"删除成功");
		}catch (Exception e){
			e.printStackTrace();
			return new Result(false,"删除失败");
		}

	}

	//按条件查询
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand tbBrand,int page,int size){
		return brandService.search(tbBrand,page,size);
	}

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//品牌控制层
app.controller('brandController' ,function($scope,$controller,brandService){
    $controller('baseController',{$scope:$scope});//继承
//读取列表数据绑定到表单中
    $scope.findAll=function(){
        brandService.findAll().success(
            function(response){
                $scope.list=response;
            }
    );
    }
        //分页
        $scope.findPage=function (page,size) {
            brandService.findPage(page,size).success(
                function(response){
                    $scope.list=response.rows;//显示当前页数据
                    $scope.paginationConf.totalItems=response.total;//更新总记录数
                }
            );
        }
        //查询单个
        $scope.findOne=function (id) {
           brandService.findOne(id).success(function (response) {

                $scope.entity=response;
            });
        }
        $scope.save=function () {
            brandService.save($scope.entity).success(
                function (response) {
                    if(response.success){
                        alert(response.message);
                        $scope.reloadList();//刷新
                    }else{
                        alert(response.message);
                    }
                }
            );
        }
        //删除选中的品牌 传递到后台 为ids的数组
        $scope.dele =function () {
            if(confirm('是否真的要删除呀')){
                brandService.dele($scope.selectIds).success(function (response) {
                    if(response.success){
                        alert('删除成功')
                        $scope.reloadList();//刷新
                    }else{
                        alert(response.message);
                    }
                });
            }
        }
        $scope.searchEntity={};
        //按条件查询页面
        $scope.search=function (page,size) {
            brandService.search(page,size,$scope.searchEntity).success(
                function (response) {
                    $scope.list=response.rows;//显示当前页数据
                    $scope.paginationConf.totalItems=response.total;//更新总记录数
                }
            );

        }
})